### ISSUE TITLE
Implement a combine,MSnExp,MSnExp method

### ISSUE THREAD

**Issue by lgatto** [lgatto/MSnbase#70]
- use `combine,AnnotatedDataFrame,AnnotatedDataFrame-method` to combine feature data (as in the `MSnSet` method); other slots should use same methods as `MSnSet` as well
- a column should also be added to the feature data to record the experiment and/or file names the respective spectra originate from
- for the assay data, if the combined feature names are unique, they could be combined and kept as is [1]
- if however there are duplicates in the combined feature names are, they should be many unique, possible using `updateFeatureNames` [1], or by adding `.1` and `.2` to the spectra of the first and second inputs.

[1]: this will probably require a `featureNames<-` for the `MSnExp` class.

**Comment by lgatto** [comment]
@sgibb Having reminded myself of the internals of `pSet`, this is not as simple as I thought. 

Simple way, where we have `combine(x, y)`:
1. rename `featureNames(x)` as `paste(featureNames(x), "1", sep = ".")` and similar for `y` with `.2`.
2. create a new assay data environment and populate it with all the renamed `x` and `y` spectra
3. rename `featureNames(featureData(x))` and `y` similarly as above and combine
4. fix the order of the combined feature data match the alphanumerical order of the new assay data
5. make the new object

Step 4 above is bad. The problem is that there is no `featureNames<-` for `MSnExp` objects, because the names in the assay data environment and feature data must be identical (same order too!). Before implementing `combine` and reordering, I think this constraint must be removed, and order is assured by the feature names of `featureData`. 

Do you have any other idea?

**Comment by sgibb** [comment]
>  because the names in the assay data environment and feature data must be identical (same order too!)

The names have to be identical. But why the order has to be identical? We can order the feature data (and rewrite `featureNames,pSet-method`) and access the assayData by `get`. There is no need to order the assayData at all (but I am not aware how much effort it would be to ensure that all selections of assayData will work this way).

**Comment by lgatto** [comment]
> But why the order has to be identical?

That was a decision made long ago. It is possible, indeed, to relax this, and would require some work (I carefully looked at the code last night). Once this is done, combining and reordering will be trivial. My only concern now is that this non-trivial amount of work might actually not be that important since your latest PR #69. I think I will still do it.

**Comment by pavel-shliaha** [comment]
Sorry, guys I read what you wrote and I still dont follow. Is there a way to reorder the spectra now?

**Comment by lgatto** [comment]
@pavel-shliaha - no, it is not possible, and is probably not advised in general, especially with the new *on disk* implementation that creates the data on-the-fly, rather than loading it in memory. In cases where this is needed, I would suggest to to the processing, filtering, ... needed, create a list of spectra with `spectra`, and the re-order that one. 

If this doesn't address your use-case, please describe it in a new issue.

**Comment by lgatto** [comment]
I am going to close this issue. With the new `OnDiskMSnExp`, it is easy and fast to load many raw files and MS levels in one go and filter from there, rather than combining. If there is a need for a `combine`, feel free to re-open.
